src/Features/CSharp/Portable/CodeCleanup/CSharpCodeCleanupService.cs

```diff
@@ -143,47 +150,76 @@ internal class CSharpCodeCleanupService : ICodeCleanupService
         }
 
-        private async Task<Document> ApplyCodeFixesAsync(Document document, DocumentOptionSet docOptions, CancellationToken cancellationToken)
+        private async Task<Document> ApplyCodeFixesAsync(
+            Document document, DocumentOptionSet docOptions,
+            IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
-            var fixAllService = document.Project.Solution.Workspace.Services.GetService<IFixAllGetFixesService>();
+            var enabledOptions = GetEnabledOptions(docOptions);
+
+            // Add a progress item for each enabled option we're going to fixup.
+            progressTracker.AddItems(enabledOptions.Length);
 
-            var dummy = new ProgressTracker();
-            foreach (var diagnosticId in GetEnabledDiagnosticIds(docOptions))
+            foreach (var (description, diagnosticIds) in enabledOptions)
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
-                var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-                var textSpan = new TextSpan(0, syntaxTree.Length);
+                progressTracker.Description = description;
+                document = await ApplyCodeFixesForSpecificDiagnosticIds(
+                    document, diagnosticIds, cancellationToken).ConfigureAwait(false);
 
-                var fixCollection = await _codeFixServiceOpt.GetFixesAsync(document, textSpan, diagnosticId, cancellationToken).ConfigureAwait(false);
-                if (fixCollection == null)
-                {
-                    continue;
-                }
+                // Mark this option as being completed.
+                progressTracker.ItemCompleted();
+            }
 
-                var fixAll = fixCollection.FixAllState;
-                var solution = await fixAllService.GetFixAllChangedSolutionAsync(
-                    fixAll.CreateFixAllContext(dummy, cancellationToken)).ConfigureAwait(false);
+            return document;
+        }
 
-                document = solution.GetDocument(document.Id);
+        private async Task<Document> ApplyCodeFixesForSpecificDiagnosticIds(
+            Document document, ImmutableArray<string> diagnosticIds, CancellationToken cancellationToken)
+        {
+            foreach (var diagnosticId in diagnosticIds)
+            {
+                document = await ApplyCodeFixesForSpecificDiagnosticId(
+                    document, diagnosticId, cancellationToken).ConfigureAwait(false);
             }
 
             return document;
         }
 
-        private IEnumerable<string> GetEnabledDiagnosticIds(DocumentOptionSet docOptions)
+        private async Task<Document> ApplyCodeFixesForSpecificDiagnosticId(
+            Document document, string diagnosticId, CancellationToken cancellationToken)
+        {
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
why getting text? syntax tree is already in memory but getting text will load text from file.

Reply:
Ah.  Good point.  I'll go back to the tree.